queue/QueueConsumer.js

```diff
@@ -24,13 +52,57 @@ module.exports = class QueueConsumer extends BaseQueue {
     if (!this._channel) {
       throw new Error('Channel not initialized')
     }
+
+    const messageContent = message.content
+
+    const maxMessageRetries = this._maxMessageRetries
+    const deadLetterQueueName = this._deadLetterQueueName
+
+    const { headers } = message.properties
+    const deadLetterHeader = headers[DEAD_LETTER_HEADER]
+
+    let rejectedRecord
+    if (deadLetterHeader) {
+      // find 'rejected' record
+      // TODO: handle other reasons in the future
+      // ex. dead letter messages from TTL expires (not implemented yet)
+      for (const record of deadLetterHeader) {
+        if (record.reason === 'rejected') {
+          rejectedRecord = record
+        }
+      }
+    }
+
+    if (rejectedRecord) {
+      // get the number of times this message has been attempted
+      const attemptCount = rejectedRecord.count
+
+      if (attemptCount < maxMessageRetries) {
```

Comment:
Should this be `<=` instead of `<`. The reason I say this is because the variable is supposed to be the number of retries not the number of total sends right? Perhaps it's just a bad variable name. If it is supposed to operate the way I think it should, we should either change the operator to `<=` or change the variable name to `maxMessageSendAttempts` or something. What do you think @charlieduong94?